x/staking/keeper/liquid_stake.go

```diff
@@ -0,0 +1,158 @@
+package keeper
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	"github.com/iqlusioninc/liquidity-staking-module/x/staking/types"
+)
+
+// SetTotalLiquidStakedTokens stores the total outstanding tokens owned by a liquid staking provider
+func (k Keeper) SetTotalLiquidStakedTokens(ctx sdk.Context, tokens sdk.Int) {
+	store := ctx.KVStore(k.storeKey)
+
+	tokensBz, err := tokens.Marshal()
+	if err != nil {
+		panic(err)
+	}
+
+	store.Set(types.TotalLiquidStakedTokensKey, tokensBz)
+}
+
+// GetTotalLiquidStakedTokens returns the total outstanding tokens owned by a liquid staking provider
+// Returns zero if the total liquid stake amount has not been initialized
+func (k Keeper) GetTotalLiquidStakedTokens(ctx sdk.Context) sdk.Int {
+	store := ctx.KVStore(k.storeKey)
+	tokensBz := store.Get(types.TotalLiquidStakedTokensKey)
+
+	if tokensBz == nil {
+		// QUESTION: Should we panic here instead?
```

Comment:
I'd prefer panic'ing over throwing an error. I actually think it's pretty [common practice ](https://github.com/cosmos/cosmos-sdk/blob/main/x/staking/keeper/delegation.go) to just panic in these keeper functions since they usually indicate that something is very wrong (different than a typical error that might arise from invalid input). 

I think the question is whether to panic or just return 0. I think I probably lean panic so we don't accidentally throw off the accounting

Reply:
in which scenarios do you see this code path getting hit? seems like if its hit there is a larger issue at hand

Reply:
plus, a zero would cause accounting errors which could be a bit dangerous if it was not detected (and would have to be detected with off chain monitoring)